scripts/build-controller-release.sh

```diff
@@ -91,9 +91,14 @@ Environment variables:
                                         custom resource definitions.
                                         Default: $K8S_RBAC_ROLE_NAME
 "
+if [ $# -lt 1 ]; then
+    echo "ERROR: $(basename "$0") needs one required parameter, the SERVICE" 1>&2
+    echo "$USAGE"
+    exit 1
+fi
 
-if [ $# -ne 2 ]; then
-    echo "ERROR: $(basename "$0") accepts exactly two parameters, the SERVICE and the RELEASE_VERSION" 1>&2
+if [ $# -gt 2 ]; then
+    echo "ERROR: $(basename "$0") accepts atmost two parameters, the SERVICE(required) and the RELEASE_VERSION(optional)" 1>&2
```

Comment:
Can we revert this change and continue using two required parameters please? We can pass `v0.0.0-non-release-version` string from wherever we call this script (in other words, the prow job...)

Reply:
> We can pass v0.0.0-non-release-version string from wherever we call this script (in other words, the prow job...)

This is currently invoked manually by service teams. Hence I was adding the default static value on their request.

Reply:
Why is this being manually invoked by service teams?

Generally, for optional parameters to our Bash scripts, we use environment variables, not arguments...

Reply:
> "Why is this being manually invoked by service teams?"

Isn't this same as service teams running `make build-controller` manually. Maybe I am missing something.
@jaypipes , Can you please elaborate on why the prowjob should be executing the build-controller-release.sh script?
(IMO, prowjobs should be testing whatever will be merged into the main branch.)

@RedbackThomson suggested to run build-controller-release.sh script as part of `make build-controller`. I think that could work here. wdyt?


> "Generally, for optional parameters to our Bash scripts, we use environment variables, not arguments..."

Sounds good. I will update this.

Reply:
I also agree that the `build-controller-release.sh` logic should remain the same, as it is required that we pass a version. When being called by developers, perhaps the `build-controller` script can pass this `v0.0.0-non-release-version`?

Reply:
@RedbackThomson , made the change to execute both scripts from make build-controller target.